### PR TITLE
BUG/TST: non-numeric EA reductions

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v3.0.0.rst` file if fixing a bug or adding a new feature.

Fixes a few bugs related to non-numeric EA reductions and expands testing by un-skipping a number of existing tests.

```
> pytest -k test_reduce pandas/tests/extension/

# main branch
2160 passed, 1222 skipped, 154 xfailed

# PR
2507 passed, 886 skipped, 143 xfailed
```

BUG fix example:

```
In [1]: import pandas as pd

In [2]: arr = pd.array([None], dtype="duration[ms][pyarrow]")

In [3]: pd.DataFrame({"a": arr}).min().dtype

# main branch
Out[3]: int64[pyarrow]

# PR
Out[3]: duration[ms][pyarrow]
```

### CODE REVIEW

File: pandas/core/arrays/string_.py [hunk 1]
```diff
@@ -525,7 +525,10 @@ def _reduce(
         self, name: str, *, skipna: bool = True, axis: AxisInt | None = 0, **kwargs
     ):
         if name in ["min", "max"]:
-            return getattr(self, name)(skipna=skipna, axis=axis)
+            result = getattr(self, name)(skipna=skipna, axis=axis)
+            if kwargs.get("keepdims"):
```
Comment (mroeschke) [inline pandas/core/arrays/string_.py:529]: Could you include `keepdims` in the `_reduce` signature instead?

File: pandas/core/arrays/datetimes.py [hunk 2]
```diff
@@ -2275,6 +2275,19 @@ def to_julian_date(self) -> npt.NDArray[np.float64]:
...
 
+    def _reduce(
+        self, name: str, *, skipna: bool = True, keepdims: bool = False, **kwargs
+    ):
+        result = super()._reduce(name, skipna=skipna, keepdims=keepdims, **kwargs)
+        if keepdims and isinstance(result, np.ndarray):
+            if name == "std":
+                from pandas.core.arrays import TimedeltaArray
+
+                return TimedeltaArray._from_sequence(result)
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:2286]: Does this possibly preserve the resolution of the input? e.g std of a ms resolution datetime array _should_ return ms resolution

Comment (lukemanley) [reply pandas/core/arrays/datetimes.py:2286]: yes, the resolution is preserved in `nanops.nanstd`.

File: pandas/core/arrays/string_.py (hunk 1 above)
Comment (lukemanley) [reply pandas/core/arrays/string_.py]: updated

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Pull request merged**